core/src/test/java/tech/tablesaw/table/TableTransposeTest.java

```diff
@@ -0,0 +1,288 @@
+package tech.tablesaw.table;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import org.junit.jupiter.api.Test;
+import tech.tablesaw.api.BooleanColumn;
+import tech.tablesaw.api.DoubleColumn;
+import tech.tablesaw.api.FloatColumn;
+import tech.tablesaw.api.IntColumn;
+import tech.tablesaw.api.LongColumn;
+import tech.tablesaw.api.StringColumn;
+import tech.tablesaw.api.Table;
+import tech.tablesaw.columns.Column;
+import tech.tablesaw.columns.numbers.DoubleColumnType;
+
+public class TableTransposeTest {
+  private static final String TABLE_NAME = "Data";
+
+  @Test
+  void transposeEmptyTable() {
+    Table empty = Table.create(TABLE_NAME);
+    Table result = empty.transpose();
+    assertEquals(empty.print(), result.transpose().print());
+    assertTableEquals(TABLE_NAME, new Object[][] {}, result);
+  }
+
+  @Test
+  void transposeDoubles() {
+    Table testTable =
+        Table.create(
+            TABLE_NAME,
+            DoubleColumn.create("value1", new double[] {1.0, 1.1, 1.2}),
+            DoubleColumn.create("value2", new double[] {2.0, 2.1, 2.2}));
+    Table result = testTable.transpose();
+
+    assertTableEquals(
+        TABLE_NAME,
+        new Object[][] {
+          {"0", 1.0, 2.0},
+          {"1", 1.1, 2.1},
+          {"2", 1.2, 2.2},
+        },
+        result);
+  }
+
+  @Test
+  void transposeWithMissingData() {
+    Table testTable =
+        Table.create(
+            TABLE_NAME,
+            DoubleColumn.create(
+                "value1", new double[] {1.0, DoubleColumnType.missingValueIndicator(), 1.2}),
+            DoubleColumn.create("value2", new double[] {2.0, 2.1, 2.2}));
+    Table result = testTable.transpose();
+
+    assertTableEquals(
+        TABLE_NAME,
+        new Object[][] {
+          {"0", 1.0, 2.0},
+          {"1", DoubleColumnType.missingValueIndicator(), 2.1},
+          {"2", 1.2, 2.2},
+        },
+        result);
+  }
+
+  @Test
+  void transposeFloats() {
```

Comment:
The test name doesn't really match what it's testing. It seems like it's testing that you can `transpose` twice and get back the original value

Reply:
Have changed the assertions to address this